patches/minecraft/net/minecraft/entity/monster/EntityZombieVillager.java.patch

```diff
@@ -8,7 +8,28 @@
      }
  
      public int func_190736_dl()
-@@ -85,6 +86,21 @@
+@@ -64,6 +65,7 @@
+     {
+         super.func_70014_b(p_70014_1_);
+         p_70014_1_.func_74768_a("Profession", this.func_190736_dl());
++        p_70014_1_.func_74778_a("ProfessionName", this.getForgeProfession().getRegistryName().toString());
+         p_70014_1_.func_74768_a("ConversionTime", this.func_82230_o() ? this.field_82234_d : -1);
+     }
+ 
+@@ -71,6 +73,12 @@
+     {
+         super.func_70037_a(p_70037_1_);
+         this.func_190733_a(p_70037_1_.func_74762_e("Profession"));
++        if (p_70037_1_.func_74764_b("ProfessionName"))
++        {
++            net.minecraftforge.fml.common.registry.VillagerRegistry.VillagerProfession p = net.minecraftforge.fml.common.registry.VillagerRegistry.instance().getRegistry().getValue(new net.minecraft.util.ResourceLocation(p_70037_1_.func_74779_i("ProfessionName")));
++            if (p == null) p = net.minecraftforge.fml.common.registry.VillagerRegistry.instance().getRegistry().getValue(new net.minecraft.util.ResourceLocation("minecraft:farmer"));
```

Comment:
Actually would be a good idea to have a constants class like everything else that uses a registry. Should talk to Mojang about that...